### DISCUSSION TITLE
skywalking oap single instance is deployed, the CPU utilization is very high

### DISCUSSION THREAD

**Discussion by gaiwey** [apache/skywalking#9121] [Q&A] [answered]
After the skywalking oap single instance is deployed, the CPU utilization is very high, reaching 200+%. Why?

JAVA_OPTS=-Xms1g -Xmx1g

![image](https://user-images.githubusercontent.com/21032881/169735775-89d35109-654b-4749-885d-87690e58553d.png)

I guess it is the frequent GC caused by the heap size of 1G, or is it caused by a large amount of calculation

**Comment by wu-sheng** [comment, marked as answer]
This is your environment, why do you ask us to guess?

**Reply by gaiwey** [reply]
Ok, what is the minimum heap size required for the oap service?

**Reply by wu-sheng** [reply]
We don't know that, JVM memory is mutating with payload.

**Reply by wu-sheng** [reply]
This is a very basic Java concept. I don't know why you think there is something we could determine directly?

**Reply by gaiwey** [reply]
Got it, I thought sw would have its own basic requirements, that's my misunderstanding

**Reply by wu-sheng** [reply]
No, we don't. All memory requirements are decided by traffic, storage chosen and performant, as well as OAP versions(usually, newer versions have better perf.)
